xds/internal/testutils/fakeclient/client.go

```diff
@@ -211,48 +210,39 @@ type ReportLoadArgs struct {
 }
 
 // ReportLoad starts reporting load about clusterName to server.
-func (xdsC *Client) ReportLoad(server string) (loadStore *load.Store, cancel func()) {
-	xdsC.loadReportCh.Send(ReportLoadArgs{Server: server})
-	return xdsC.loadStore, func() {}
+func (c *Client) ReportLoad(server string) (loadStore *load.Store, cancel func()) {
+	c.loadReportCh.Send(ReportLoadArgs{Server: server})
+	return c.loadStore, func() {}
 }
 
 // LoadStore returns the underlying load data store.
-func (xdsC *Client) LoadStore() *load.Store {
-	return xdsC.loadStore
+func (c *Client) LoadStore() *load.Store {
+	return c.loadStore
 }
 
 // WaitForReportLoad waits for ReportLoad to be invoked on this client and
 // returns the arguments passed to it.
-func (xdsC *Client) WaitForReportLoad(ctx context.Context) (ReportLoadArgs, error) {
-	val, err := xdsC.loadReportCh.Receive(ctx)
+func (c *Client) WaitForReportLoad(ctx context.Context) (ReportLoadArgs, error) {
+	val, err := c.loadReportCh.Receive(ctx)
 	return val.(ReportLoadArgs), err
 }
 
-// Close closes the xds client.
-func (xdsC *Client) Close() {
-	xdsC.closeCh.Send(nil)
-}
-
-// WaitForClose waits for Close to be invoked on this client and returns
-// context.DeadlineExceeded otherwise.
-func (xdsC *Client) WaitForClose(ctx context.Context) error {
```

Comment:
This isn't right.

The actual question is, why is this method not used in any tests (it should be all the tests, but not only the resolver tests would matter).

The right fix is to call and check for this in the xds resolver tests, not to delete this.

Reply:
> The actual question is, why is this method not used in any tests (it should be all the tests, but not only the resolver tests would matter).

I had assumed this was once here for some kind of synchronization, then the usage was removed because it wasn't necessary.

This doesn't seem like the right way to test that the resolver closes the client.  We should be ensuring that by the time `xdsResolver.Close()` returns, `xdsClient.Close()` has been called -- not "block until it happens".

I can add this test case in another PR immediately after this if you want.

Reply:
#4509

Reply:
Same as the other comment. This change should be moved to the other PR.

Reply:
I'll just close this PR and open a series of other PRs, one for each change.